src/iop/Permutohedral.h

```diff
@@ -83,8 +83,8 @@ template <int KD, int VD> class HashTablePermutohedral
       for (int i = 0; i < KD; i++) { if (key[i] != other.key[i]) return false; }
       return true;
     }
-    unsigned hash;      // cache the hash value for this key
-    short key[KD];      // key is a KD-dimensional vector
+    unsigned hash{ 0 };      // cache the hash value for this key
+    short key[KD]{};      // key is a KD-dimensional vector
```

Comment:
These two lines ensure that the member variables get properly initialized via the default constructor.

Reply:
Would be nice to keep the comment aligned.

Reply:
I completely agree. Indenting and formatting of the whole file are in a bad shape. I would like to open a 2nd pull request that is about reformatting only once this one has been merged. Code changes plus heavy reformatting renders review a difficult task.

Reply:
Why not just add a second commit on this PR ? I do that all the time, a commit for style changes and commits for the fix or new feature.

Reply:
@TurboGit If splitting the pull request into two commits as suggested is an acceptable option I would be happy to do so. This simplifies a lot.

Reply:
Yes, a PR can have multiple commits, as much as it is needed to make the review easier.